shiny/express/layout.py

```diff
@@ -225,6 +232,111 @@ def layout_column_wrap(
+          first element, 8 columns to the second element, and 12 columns to the third
+          element (which wraps to the next row). Negative values are also allowed, and
+          are treated as empty columns. For example, `col_widths=(-2, 8, -2)` would
+          allocate 8 columns to an element (with 2 empty columns on either side).
+        * A dictionary of column widths at different breakpoints. The keys should be
+          one of `"xs"`, `"sm"`, `"md"`, `"lg"`, `"xl"`, or `"xxl"`, and the values are
+          either of the above. For example, `col_widths={"sm": (3, 3, 6), "lg": (4)}`.
+
+    row_heights
+        The heights of the rows, possibly at different breakpoints. Can be one of the
+        following:
+
+        * A numeric vector, where each value represents the
+          [fractional unit](https://css-tricks.com/introduction-fr-css-unit/)
+          (`fr`) height of the relevant row. If there are more rows than values
+          provided, the pattern will be repeated. For example, `row_heights=(1, 2)`
+          allows even rows to take up twice as much space as odd rows.
+        * A list of numeric or CSS length units, where each value represents the height
+          of the relevant row. If more rows are needed than values provided, the pattern
+          will repeat. For example, `row_heights=["auto", 1]` allows the height of odd
+          rows to be driven my it's contents and even rows to be
+          [`1fr`](https://css-tricks.com/introduction-fr-css-unit/).
+        * A single string containing CSS length units. In this case, the value is
+          supplied directly to `grid-auto-rows`.
+        * A dictionary of row heights at different breakpoints, where each key is a
+          breakpoint name (one of `"xs"`, `"sm"`, `"md"`, `"lg"`, `"xl"`, or `"xxl"`)
+          and where the values may be any of the above options.
+
+    fill
+        Whether or not to allow the layout to grow/shrink to fit a fillable container
+        with an opinionated height (e.g., :func:`~shiny.ui.page_fillable`).
+
+    fillable
+        Whether or not each element is wrapped in a fillable container.
+
+    gap
+        Any valid CSS unit to use for the gap between columns.
+
+    class_
+        CSS class(es) to apply to the containing element.
+
+    height
+        Any valid CSS unit to use for the height.
+
+    **kwargs
+        Additional attributes to apply to the containing element.
+
+    Returns
+    -------
+    :
+        An :class:`~htmltools.Tag` element.
+
+    See Also
+    --------
+    * :func:`~shiny.express.layout.layout_column_wrap` for laying out elements into a
+      uniform grid.
+
+    Reference
+    --------
+    * [Bootstrap CSS Grid](https://getbootstrap.com/docs/5.3/layout/grid/)
+    * [Bootstrap Breakpoints](https://getbootstrap.com/docs/5.3/layout/breakpoints/)
+    """
+    return RecallContextManager(
+        ui.layout_columns,
+        default_page=page_fillable(),
```

Comment:
I suppose this is consistent with `layout_column_wrap()`, but it's likely going to get removed for an alternative approach to detecting/setting intelligent page defaults, it'd be a heads-up move to remove this here and `layout_column_wrap()` 

```suggestion
```

Reply:
Sounds good, I removed this line! I'd like to keep the changes in this PR limited to the addition of `layout_columns()`. I'll be happy to follow up with a PR for `layout_column_wrap()`.